From 25138c9fe9d64970e27be0b0ec3f01ef04c687d6 Mon Sep 17 00:00:00 2001
From: Paul Bovbel <paul@bovbel.com>
Date: Mon, 1 Feb 2021 15:44:55 -0500
Subject: [PATCH] Add install rules

---
 CMakeLists.txt | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 94fe3bc..7c81eb6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,6 +20,14 @@ include(OptionRequires)
 include(CTest)
 include(CheckIncludeFileCXX)
 
+set(VRPN_BUILD_CLIENT_LIBRARY TRUE)
+set(VRPN_BUILD_JAVA FALSE)
+set(VRPN_BUILD_PYTHON FALSE)
+set(VRPN_BUILD_CLIENTS FALSE)
+set(VRPN_BUILD_SERVERS FALSE)
+set(VRPN_BUILD_SERVER_LIBRARY FALSE)
+set(BUILD_TESTING FALSE)
+
 if((NOT CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR) OR VRPN_SUBPROJECT_BUILD)
 	# If you're using this as a subproject and want things installed, set VRPN_INSTALL TRUE.
 	set(SUBPROJECT TRUE)
@@ -1327,6 +1335,19 @@ if(VRPN_BUILD_SERVER_LIBRARY)
 	add_cppcheck(vrpnserver STYLE UNUSED_FUNCTIONS)
 endif()
 
+# Install catkin package.xml
+install(FILES package.xml DESTINATION share/vrpn)
+
+install(
+	FILES cmake/FindVRPN.cmake
+	DESTINATION share/vrpn/cmake
+	RENAME VRPNConfig.cmake)
+
+install(
+	FILES cmake/Findquatlib.cmake
+	DESTINATION share/quatlib/cmake
+	RENAME quatlibConfig.cmake)
+
 
 if(VRPN_BUILD_CLIENT_LIBRARY)
 	add_library(vrpn ${VRPN_CLIENT_SOURCES} ${VRPN_CLIENT_PUBLIC_HEADERS})
-- 
2.17.1

